internal/manifests/gateway/openshift.go

```diff
@@ -86,23 +86,37 @@ func clusterRoleBinding(tempo v1alpha1.TempoStack) *rbacv1.ClusterRoleBinding {
 
 func route(tempo v1alpha1.TempoStack) *routev1.Route {
 	labels := manifestutils.ComponentLabels(manifestutils.GatewayComponentName, tempo.Name)
+
+	var tlsCfg *routev1.TLSConfig
+	switch tempo.Spec.Template.Gateway.Ingress.Route.Termination {
+	case v1alpha1.TLSRouteTerminationTypeInsecure:
+		// NOTE: insecure, no tls cfg.
+	case v1alpha1.TLSRouteTerminationTypeEdge:
+		tlsCfg = &routev1.TLSConfig{Termination: routev1.TLSTerminationEdge}
+	case v1alpha1.TLSRouteTerminationTypePassthrough:
+		tlsCfg = &routev1.TLSConfig{Termination: routev1.TLSTerminationPassthrough}
+	case v1alpha1.TLSRouteTerminationTypeReencrypt:
+		tlsCfg = &routev1.TLSConfig{Termination: routev1.TLSTerminationReencrypt}
+	default: // NOTE: if unsupported, end here.
+		return nil
```

Comment:
Should we log something in this case? Because, for me, it sounds like failing silently

Reply:
`TLSRouteTerminationType` is an enum in the CRD, so this case should never happen in theory.
Unless we update the list of valid enum values and forget to update this switch, so I guess yep, it makes sense to print a log message (could you add the same log message to the `switch` in the query_frontend also?).

Reply:
It should be checked by link https://github.com/grafana/tempo-operator/blob/7c7ee6470cd01138f1225d8734dd7f3e7229135f/.golangci.yaml#L29